### PR TITLE
Add Input to palette

### PR DESCRIPTION
**Pull request by jonallured**
This PR adds the Input component as speced out over on #253. Here are a couple shots:

Screen Shot 2019-03-21 at 11 28 44 AM
Screen Shot 2019-03-21 at 11 28 55 AM

The aim of this component is to replace the one in Reaction:

https://github.com/artsy/reaction/blob/e062da3ab478ebdbce4e3a85812ee2d7be1bb8a0/src/Components/Input.tsx

And the one in Volt:

https://github.com/artsy/volt/blob/ceda8118dd520525545acf3059337201c0a7f7cc/app/javascript/shared/form/components/TextInput.jsx

But that work will happen after merging here and then it'll be possible that I'll need follow up PRs to tweak things for them. My plan would be to get this into master, upgrade clients and only then start migrating them over to this one as it makes sense.

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-palette* ready!

Built with commit 562824b8021ce6601699a0a96829286094611547

https://deploy-preview-339--artsy-palette.netlify.com

File: packages/palette/src/elements/Input/Input.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+export const Input: React.SFC<InputProps> = ({
+  description,
+  disabled,
+  error,
+  required,
+  title,
+  ...rest
+}) => {
+  return (
+    <StyledBox>
```
Comment (damassi) [inline packages/palette/src/elements/Input/Input.tsx]: Maybe just `<Box width='100%'>`?

File: packages/palette/src/elements/Input/Input.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        </Sans>
+      )}
+    </StyledBox>
+  )
+}
+
+const StyledBox = styled(Box)`
+  width: 100%;
+`
+StyledBox.displayName = "StyledBox"
```
Comment (damassi) [inline packages/palette/src/elements/Input/Input.tsx]: Needed? If so, it might be a little more semantic to use something like `<InputContainer>`

**Comment by damassi** [comment]
@jonallured Could we add a few light tests around the input? Just something structural that asserts that certain fields appear if certain props are passed. Otherwise, LGTM 👍

**Comment by jonallured** [comment]
@damassi good feedback as always!! I took an honest shot at testing this sucker - let me know how you feel about this coverage.

**Comment by damassi** [comment]
Looks great 👍

merge on green

**Comment by jonallured** [comment]
@damassi I took off MOG so that @jeffreytr could have a chance to do a QA pass prior to merging, thanks!

**Comment by jeffreytr** [comment]
Everything looks good on my end!!

**Comment by artsyit** [comment]
:rocket: PR was released in v4.4.0 :rocket:

**Pull request merged**